web/src/user_profile.ts

```diff
@@ -511,6 +511,9 @@ export function show_user_profile(user: User, default_tab_key = "profile-tab"):
     const profile_data = realm.custom_profile_fields
         .flatMap((f) => get_custom_profile_field_data(user, f, field_types) ?? [])
         .filter((f) => f.name !== undefined);
+    original_values = {
+        user_id: user.user_id.toString(),
+    };
```

Comment:
Can you explain what the impact would be of adding other fields here?

Reply:
When updating the fields, we checked the user_id from the original values. The bug was that original values were undefined since it wasn't being set initially when we opened the user profile. When we open the edit form directly, we use `dialog_widget` to get the data which fetches all the data from the form including full_name, email, custom_profile_feilds etc. 
We only need `user_id` to compare if the updates coming are really for this particular user_id or not as per the conditionals. I believe it doesn't matter if we add other fields initially since we aren't using it to update the data in user_profile_modal.